include/forcing/DataProviderSelectors.hpp

```diff
@@ -121,6 +121,12 @@ class CatchmentAggrDataSelector
     std::string id_str; //< the catchment to access data for
 };
 
+
+#if !defined(__APPLE__) && !defined(__llvm__) && defined(__GNUC__) && __GNUC__ < 8
+// Enable a workaround for old GCC versions (but not Clang or Apple's fake-GCC that's actually Clang)
+#define NGEN_SELECTOR_CAST
```

Comment:
Out of curiosity did this set of changes fix an issue being reported by the ASAN tests or is it just a target of opportunity to make cleaner?

Reply:
Target of opportunity - we knew it was wrong, and I was seeing a warning in compiler output over and over as I was pulling this together.